src/main/java/com/hms_networks/americas/sc/thingworx/TWConnectorMain.java

```diff
@@ -214,6 +219,14 @@ else if (queueDoubleDataRateEnabled && queueBehindMillis == 0) {
       } catch (Exception e) {
         Logger.LOG_CRITICAL("An error occurred while reading data from the historical log.");
         Logger.LOG_EXCEPTION(e);
+        // Chances are that trying the exact same read next time will result in the same exception,
+        // so advance the start time tracking
+        try {
+          HistoricalDataQueueManager.advanceTrackingStartTime();
+        } catch (Exception e1) {
+          Logger.LOG_CRITICAL("An error occurred while attempting  to advance the historical log.");
+          Logger.LOG_EXCEPTION(e1);
+        }
```

Comment:
There should probably be a log that the section of data is being skipped.

Reply:
good call.  will change.